mir-module/src/main/resources/xsl/response-facets.xsl

```diff
@@ -0,0 +1,232 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                xmlns:encoder="xalan://java.net.URLEncoder"
+                xmlns:i18n="xalan://org.mycore.services.i18n.MCRTranslation"
+                xmlns:mcrxsl="xalan://org.mycore.common.xml.MCRXMLFunctions"
+                xmlns:xalan="http://xml.apache.org/xalan"
+                xmlns:fn="http://www.w3.org/2005/xpath-functions"
+                exclude-result-prefixes="i18n mcrxsl encoder xalan fn">
+
+  <xsl:param name="MIR.Response.Facet.Prefix.Classification"/>
+
+  <xsl:template name="facets">
+
+    <xsl:for-each select="/response/lst[@name='facet_counts']/lst[@name='facet_fields']/*">
+      <xsl:variable name="facet_name" select="self::node()/@name"/>
+      <xsl:variable name="classId">
+        <!-- Check if facet has a prefix -->
+        <xsl:choose>
+          <xsl:when test="contains($facet_name, $MIR.Response.Facet.Prefix.Classification)">
+            <xsl:value-of select="substring-after($facet_name, $MIR.Response.Facet.Prefix.Classification)"/>
+          </xsl:when>
+          <xsl:otherwise>
+            <xsl:value-of select="$facet_name" />
+          </xsl:otherwise>
+        </xsl:choose>
+      </xsl:variable>
+
+      <!-- Get the classification named $classId -->
+      <xsl:variable name="classification">
+        <xsl:value-of select="concat('notnull:classification:metadata:all:children:',$classId)"/>
```

Comment:
One should check beforehand if value is a classification otherwise the log will be messed up with stacktraces produced by the notnull-uri resolver

Reply:
The variable $classification is already checked in the following places:

```
                    <xsl:when test="name(document($classification)/*) != 'null'">
                      <xsl:variable name="category_label" select="document($classification)/mycoreclass/label[@xml:lang=$CurrentLang]/@text"/>
```
...

```
                <xsl:when test="name(document($classification)/*) != 'null'">
                  <xsl:variable name="category" select="@name"/>
```